datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -107,6 +110,7 @@ impl<S: SimplifyInfo> ExprSimplifier<S> {
             info,
             guarantees: vec![],
             canonicalize: true,
+            max_simplifier_iterations: DEFAULT_MAX_SIMPLIFIER_ITERATIONS,
```

Comment:
Right now there's no reason to have a struct field, but the idea is that we might want to make this configurable later.

Reply:
Perhaps we could have an API to update it to mirror

```rust
    pub fn with_canonicalize(mut self, canonicalize: bool) -> Self {
        self.canonicalize = canonicalize;
        self
    }
```

Something like 
```rust
    pub fn with_max_simplifier_iterations(mut self, max_simplifier_iterations: usize) -> Self {
        self.max_simplifier_iterations = max_simplifier_iterations;
        self
    }
```

Perhaps